docs/en/latest/back-end-tests.md

```diff
@@ -0,0 +1,98 @@
+
+## Writing E2E Tests
+
+Currently, the backend of apisix-dashboard have two types of e2e tests. One is plain e2e, the other is e2enew, where in the first one, tests are written using Go's built-in, native testing package, for the later, the tests are grouped into test suites and are evaluated using [ginkgo](https://onsi.github.io/ginkgo/) - a testing framework which helps in writing more expressive tests such that reading and writing tests give a pleasant experience.
+
+**Slowly, we are migrating all of our e2e tests to e2enew. So it is always recommended to write any new tests using ginkgo unless some situation arises. In such cases, please discuss your concerns with the community.
+
+For value assertion, we are using the [assert](https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/assert) package by testify. It provides lots of easy to use functions for assertion where the first argument is   `*testing.T` object which you can obtain from the used framework for testing. For the built-in testing package, each test cases have this as the first argument of the test itself. For ginkgo `ginkgo.GinkgoT()` returns the mentioned object.
+
+If you are creating any test which requires making HTTP calls to any of the following node which involves `manager-api` or `apisix`, after setting up the environment (please refer [`backend-e2e.md`](./back-end-e2e.md) for the details), you can use the `HttpTestCase` struct which provides a nice interface to make the calls along with checking the response. Here's a brief description of the most used fields of the struct,
+
+```go
+type HttpTestCase struct {
+    Desc          string                // Description about the test case.
+    Object        *httpexpect.Expect    // returns a httpexpect object i.e. on which host the request is going to be made.
+    Method        string                // HTTP request methods ( GET, POST, PATCH, PUT, DELETE, OPTIONS).
+    Path          string                // the route path of that host
+    Query         string                // Query params
+    Body          string                // The request Body. Commonly used in POST, PUT, PATCH.
+    Headers       map[string]string     // Request headers. Include authorization header for secure routes.
+    ExpectStatus  int                   // Expected HTTP status code from the response
+    ExpectCode    int                   // Code generated by the host. Generally 0 for http.StatusOK.
+    ExpectMessage string                // The response message provided in the response by the host.
+    ExpectBody    interface{}           // The expected message body as a response.
+    Sleep         time.Duration //ms    // Cooldown period before making next request.
+}
+```
+
+Now to run a test use `RunTestCase(tc HttpTestCase)` or `testCaseCheck(tc HttpTestCase, t *testing.T)` for the `e2enew` or `e2e` respectively.
```

Comment:
At present, we are doing the existing E2E test to migrate to ginkgo. IMHO, it's better to add a note here to prompt users to write E2E tests using ginkgo as much as possible. After the migration is completed, the E2E test method can be removed here.

Reply:
Hii. I had added the note in line [48](https://github.com/apache/apisix-dashboard/pull/1625/files#diff-c6ae044f165581d7c8c5df0dac40b063a1461f412acf700ea02c65ed06faa176R47-R49). Would that be sufficient or should I add it here too explicitly?
Thanks

Reply:
No problem. Can we just write the ginkgo test method?
If E2E is used, the old method still needs to be migrated.
cc @starsz @nic-chen @bisakhmondal  Do you have any good suggestions？

Reply:
IMO, we can even not describe e2e test. Only introduce how to write ginkgo test.
For that we are migrating all of our e2e tests to e2enew.

Reply:
> IMO, we can even not describe e2e test. Only introduce how to write ginkgo test.
> For that we are migrating all of our e2e tests to e2enew.

+1 , we could only introduce how to write ginkgo test.

Reply:
+1 Add a note here， only accept ginkgo test in new PR

Reply:
Nice!! Making the necessary changes then :)